### PR TITLE
Login

### PR DESCRIPTION
**Pull request by Rockyraven**
added guest login button
data fetch from backen
loginpage
d

### CODE REVIEW

File: src/context/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+const { createContext, useState, useContext } = require("react");
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [encodedToken, setEncodedToken] = useState(null);
```
Comment (sonishreyas) [inline src/context/AuthContext.js:9]: Instead of separate states you can have these as useReducer

**Review by sonishreyas — APPROVED** [review]
The code looks good and easy to understand. I have left a suggestion rest it looks good.

File: src/context/AuthContext.js (hunk 1 above)
Comment (Rockyraven) [reply src/context/AuthContext.js:9]: ok I will do that

**Review by Shubhamkr2610 — APPROVED** [review]
rest everything looks good to me . after changing the suggestion that other folks suggested you are good to go

**Pull request merged**